### PR TITLE
1004 tooltips disabled buttons

### PR DESCRIPTION
**Pull request by Fabian5150**
Closes https://github.com/corona-school/project-user/issues/1004

Lots of small stuff

Added generalized Component components/DisablebleButton to automatically render a tooltip around a button if disabled.
This component requires a "reasonDisabled" field so all Buttons, that can be disabled (assuming they use this new component) have provide one.
This way the other components stay a bit cleaner and more unified
Than changed all accuring Buttons, that can be disabled to use this component.
Other button props now have to be assigned slightly different, all as one object given to DisablebleButton, but it does not sacrifice any functionality or creates significantly more effort/complexitiy.

### CODE REVIEW

File: src/components/DisablebleButton.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+type ButtonProps = ComponentProps<typeof Button>;
+
+type DisablebleButtonProps = {
+    isDisabled: boolean;
+    reasonDisabled: string;
+    buttonProps?: ButtonProps;
+    disabledButtonContent?: ReactNode;
+};
+
+const DisablebleButton: React.FC<PropsWithChildren<DisablebleButtonProps>> = ({ isDisabled, reasonDisabled, disabledButtonContent, children, buttonProps }) => {
```
Comment (Jonasdoubleyou) [inline src/components/DisablebleButton.tsx:13]: DisableableButton?

File: src/components/DisablebleButton.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Button, Tooltip } from 'native-base';
+import { PropsWithChildren, ComponentProps, ReactNode } from 'react';
+
+type ButtonProps = ComponentProps<typeof Button>;
+
+type DisablebleButtonProps = {
```
Comment (Jonasdoubleyou) [inline src/components/DisablebleButton.tsx:6]: = { ... } & ButtonProps; and then pass them on as spread properties (...rest in parameters, {...rest} in component)? Or any reason to have "buttonProps" on a component named ...Button?

**Review by Jonasdoubleyou — COMMENTED** [review]
nice refactoring, some minor improvements to be made to the button component

File: src/components/DisablebleButton.tsx (hunk 1 above)
Comment (Fabian5150) [reply src/components/DisablebleButton.tsx:13]: I don't really like the name either... But I couldn't come up with something better, I think just "DisabledButton" or something like that would give a wrong impression about what the component is actually there for

File: src/components/DisablebleButton.tsx (hunk 2 above)
Comment (Fabian5150) [reply src/components/DisablebleButton.tsx:6]: Should've known that earlier... Gave up to earlier on finding a better way to pass the props down to the button, but it's in the new commit now. :)

File: src/components/DisablebleButton.tsx (hunk 1 above)
Comment (Fabian5150) [reply src/components/DisablebleButton.tsx]: Oh... will fix that real quick here

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**